packages/data-stores/src/onboard/constants.ts

```diff
@@ -24,4 +24,5 @@ export enum SiteIntent {
 	ReadyMadeTemplate = 'readymade-template',
 	AIAssembler = 'ai-assembler',
 	Newsletter = 'newsletter',
+	FreePostSetup = 'free-post-setup',
```

Comment:
This value was previously hardcoded for passing into the Tracks event: https://github.com/Automattic/wp-calypso/pull/93178/files#diff-0f11d3ee85ff317ba5220eca7c530427e95b7760defd1451df07464662d5356bL24

Perhaps this was intentional? 

Assuming we'd want `SiteIntent` to reflect the intent recorded in the site's meta (e.g. the return value from [useSiteIntent](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/queries/use-site-intent.ts)), which is typically the name of the flow creating the site, then adding `free-post-setup` here may not be correct - if a site is not created through this flow in the end.

Reply:
Not sure who to ping about this. It looks like all of @edanzer @bstoynov @jeyip were involved in the preceding work https://github.com/Automattic/wp-calypso/pull/70908 

cc @daledupreez who introduced the `isSignupFlow` prop - in case you know more about this flow?

Reply:
I am not sure as to how the value is used. I can say that `isSignupFlow` should control whether we create `calypso_signup_start` events when users start the flow. For this flow, it's an onboarding flow where we don't create a site (we operate on an existing one), so we don't want to trigger `calypso_signup_start`.

That's way less relevant as to what we're doing with that constant. AFAICT, we should be able to use the flow name here.

Reply:
I think my main concern is how the two concepts (`intent` used here and `intent` used in the site store) aren't mapped to the same type. I can see the `SiteIntent` in the onboard store was until recently a `SiteGoal`, before that a `SiteKey`, etc. and there's a bit of arbitrariness in what's included here e.g. not sure if any of `Write`, `Sell`, `Build` are anything more than frontend/onboarding confers, not to be recorded through `/sites/:id/site-intent` endpoint.

Classic used the same concept through a Step that would record on `/sites/:id/site-intent` - similarl to what is done via the data-store `site.setIntentOnSite` action. It's more sporadic/random, as anything can call this from anywhere.

So I think some ambiguity will exist until `setIntentOnSite` and `SiteIntent` map over the same correct values. It might help to centrally call `setIntentOnSite` in the framework once a site gets created (if not already). It may not be as easy - I can see random usages, also non-signup flows calling this to "update". It might be worth looking into.

Reply:
Added to issue: https://github.com/Automattic/wp-calypso/issues/93242

It's something to follow up on. For now, I intend to keep whatever was recorded to be recorded as previously. The risk may be adding more values to `SiteIntent` that may never get recorded as site intent. Not sure how else to have a concrete type for the Tracks recorders without over-engineering around ti.